File: problems/1746-largest-substring-between-two-equal-characters/largest-substring-between-two-equal-characters.py
class Solution:
    def maxLengthBetweenEqualCharacters(self, s: str) -> int:
        maxDiff = -1
        if len(s) == 1:
            return maxDiff

        hashMap = {}
        for i in range(len(s)):
            if s[i] in hashMap.keys():
                maxDiff = max(maxDiff, i - hashMap[s[i]] - 1)
            else:
                hashMap[s[i]] = i
        return maxDiff

File: problems/1746-largest-substring-between-two-equal-characters/largest-substring-between-two-equal-characters.js
/**
 * @param {string} s
 * @return {number}
 */
const maxLengthBetweenEqualCharacters = (s) => {
    // only one string
    if (s.length === 1) return -1;
    // hashmap for indecies
    const hashMap = {}
    for (let i = 0; i < s.length; i++) {
        if (hashMap[s[i]]) {
            hashMap[s[i]].push(i);
        } else {
            hashMap[s[i]] = [i];
        }
    }
    
    let maxDiff = -Infinity;
    for (const [key, value] of Object.entries(hashMap)) {
        if (value.length > 1) {
            const tmp = (value[value.length - 1]) - (value[0] + 1)
            maxDiff = Math.max(maxDiff, tmp);
        }
    }
    return maxDiff >= 0 ? maxDiff : -1;
};